course_catalog/etl/deduplication.py

```diff
@@ -39,3 +44,71 @@ def get_most_relevant_run(runs):
             most_relevant_run = next((run for run in runs.reverse()))
 
     return most_relevant_run
+
+
+def generate_duplicates_yaml():
+    "Generate updated list of mitx duplicates"
+
+    platform = PlatformType.mitx.name
+    duplicates_url = settings.DUPLICATE_COURSES_URL
+    output = {}
+
+    if duplicates_url is not None:
+        response = requests.get(duplicates_url)
+        response.raise_for_status()
+        existing_duplicates_for_all_platforms = yaml.safe_load(response.text)
+    else:
+        existing_duplicates_for_all_platforms = {}
+
+    if platform in existing_duplicates_for_all_platforms:
+        desired_course_ids_from_existing_duplicate_file = [
+            group["course_id"]
+            for group in existing_duplicates_for_all_platforms[platform]
+        ]
+    else:
+        desired_course_ids_from_existing_duplicate_file = []
+
+    duplicate_titles = (
+        Course.objects.values("title")
+        .annotate(Count("id"))
+        .filter(id__count__gt=1)
+        .filter(platform=platform)
+    )
+    duplicate_courses = (
+        Course.objects.filter(title__in=[item["title"] for item in duplicate_titles])
+        .annotate(Count("offered_by"))
+        .values("course_id", "title", "offered_by__count")
+        .order_by("title")
+    )
+    duplicate_courses = [
```

Comment:
Can we rename this variable to `duplicate_course_groups`, and leave the other variable as `duplicate_courses`?